user-console/src/main/resources/org/pentaho/mantle/public/home/js/home.js

```diff
@@ -85,6 +85,32 @@ define([
       }
     });
 
+    $('#btnCreateNew').on('shown.bs.modal', function () {
+      $('.popover').find("button").first().focus();
+      $('.popover .btn').keydown(function (event) {
+        if (event.shiftKey && event.keyCode === 9) {
+          var firstButtonId = $('.popover .btn').first().attr("id");
+          if (firstButtonId === this.id) {
+            $('#btnCreateNew').popover('hide');
+            $('#btnCreateNew').focus();
+            event.preventDefault();
+            event.stopPropagation();
+          }
+        } else if (event.keyCode === 27) {
+          $('#btnCreateNew').popover('hide');
+          $('#btnCreateNew').focus();
+        } else if (event.keyCode === 9) {
```

Comment:
I'm not too familiar with javascript coding standards. I'll accept push back on this...

all the `event.keyCode === X`  have magic numbers: https://codehs.com/glossary/term/14

I would either declare a constant or write a comment by each one stating the meaning of each number.

1. it easier to quickly read without referencing external documentation.
2. based off Mozilla https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode#constants_for_keycode_value this feature might be deprecated, so better to document what the original author intended as well. Some browsers might not break compatibility in the future.

Reply:
I agree, its meaningful to add what that number means as well. I added the comments as you suggested.